### PR TITLE
Add S1G in suggested fields

### PR DESCRIPTION
**Pull request by WFA-achuang**
Add S1G PHY information in radiotap

### CODE REVIEW

File: fields/S1G.md [hunk 1]
```diff
@@ -0,0 +1,48 @@
+---
+title: S1G
+categories: [suggested]
+---
+Type
+: 32 
+
+Structure
+: u16 known, u16 data1, u16 data2,
```
Comment (jmberg) [inline fields/S1G.md]: shouldn't have a comma there

File: fields/S1G.md [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+| **`0x00C0`** | Number Spatial Streams: 0=1 Spatial stream, 1=2, .. 3=4 Spatial streams |
+| **`0x0F00`** | Bandwidth: 0=1MHz, 1=2MHz, 2=4MHz, 3=8MHz, 4=16MHz, 5-15 reserved |
+| **`0xF000`** | MCS (MCS rate index, 0-10, 11-15 reserved) |
+
+## data2
+
+| **`0x0007`** | Color: 0-7 |
+| **`0x0008`** | Uplink indication |
+| **`0x00F0`** | Reserved |
+| **`0xFF00`** | RSSI |
```
Comment (jmberg) [inline fields/S1G.md]: there's no 'known' bit for the RSSI, but it shouldn't be needed anyway? there should be a separate "Antenna signal" for it, no?

Comment (WFA-achuang) [reply fields/S1G.md]: Yes, there is no "known" bit for RSSI. These definition is finalized and test during program announcement. I just post the definition based on current implementation.

Comment (jmberg) [reply fields/S1G.md]: I understand that, but that doesn't mean it's good?

This is what you get for not following the right process I guess.

Personally, I think the official version shouldn't even carry this field.

Also, come to think of it, the process says there should be a producer we can look at ...

I suppose it's better to have this discussion on the mailing list.

Comment (WFA-achuang) [reply fields/S1G.md]: Each frame during driver capture should always have RSSI value. So we should not need "known" bit

Comment (jmberg) [reply fields/S1G.md]: That's maybe true for your implementation now, but it shouldn't be _required_ of an implementation?

And anyway, the RSSI is usually presented as the "Antenna signal" field, why not here?

Comment (WFA-achuang) [reply fields/S1G.md]: Let me check the sample sniffer and confirm if there is "Antenna signal" field.

Comment (jmberg) [reply fields/S1G.md]: That kind of seems the wrong way around ... this discussion should've happened years ago, and we *did* have this discussion (with Richard), but nobody followed up on your side.

Comment (jmberg) [reply fields/S1G.md]: There actually is a possible difference between signal and RSSI, but here you don't even really define what RSSI means, and even in that case it should probably not be S1G specific, but another field.

Comment (WFA-achuang) [reply fields/S1G.md]: I check the proposal again in https://lore.kernel.org/radiotap/ and our implementation. I also confirmed the radiotap headers driver provided in S1G sniffer capture. I will only keep the consistent definition.

Comment (WFA-achuang) [reply fields/S1G.md]: Do I need to remove Uplink indication? the field is not in the proposal.

Comment (WFA-achuang) [reply fields/S1G.md]: I also removed uplink indication to sync with the original proposal.
Please help to check if there is any problem.

**Comment by WFA-achuang** [comment]
Hi,
Because the change was submitted into Wireshark upstream several years ago, the change is to sync between radiotap definition and Wireshark. The main purpose is to reserve radiotap ID 32 for S1G.
I already removed several fields and only keep necessary fields. Please let me know what is the main problem in the PR because this PR is directly closed?

**Comment by jmberg** [comment]
I _thought_ I added the field as parsed by wireshark ... http://www.radiotap.org/fields/S1G.html

If I got it wrong, then anyway the PR wouldn't have worked any more, could you then please send an update?

**Comment by WFA-achuang** [comment]
Thanks for adding S1G fields as parsed by Wireshark.
I don't have any problem about S1G.

**Pull request closed without merging**